jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2372,7 +2376,92 @@ else if (oldQueryParams == null || oldQueryParams.size() == 0)
     @Override
     public <T extends HttpUpgradeHandler> T upgrade(Class<T> handlerClass) throws IOException, ServletException
     {
-        throw new ServletException("HttpServletRequest.upgrade() not supported in Jetty");
+        try
```

Comment:
Should we check if the response is committed and if not do a flushBuffers to commit. 
Alternatively, if flushBuffers doesn't work, we could send the 101 the same as we send 102s in `Response.sendProcessing`

Probably should also check that response body is not written.

Reply:
I'm not sure about this one. Let's discuss it if you still think it's needed after you reviewed my latest changes.

Reply:
I would still like to see a definitely commit by sending the 101 response at this point.   If we use the same mechanism as Response.sendProcessing, then I think we might need less changes in the generator (but not sure on that).

Eitherway, seeing a clean commit point would make the code more understandable.

Reply:
I modified the impl according to our discussion on the matter.